bark/models/dynamic/single_track_steering_rate.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2020 fortiss GmbH
+//
+// Authors: Julian Bernhard, Klemens Esterle, Patrick Hart and
+// Tobias Kessler
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+
+#ifndef BARK_MODELS_DYNAMIC_SINGLE_TRACK_STEERING_RATE_HPP_
+#define BARK_MODELS_DYNAMIC_SINGLE_TRACK_STEERING_RATE_HPP_
+
+#include <algorithm>
+#include <memory>
+#include "bark/commons/transformation/frenet_state.hpp"
+#include "bark/models/dynamic/dynamic_model.hpp"
+
+namespace bark {
+namespace models {
+namespace dynamic {
+
+
+class SingleTrackSteeringRateModel : public DynamicModel {
+ public:
+  explicit SingleTrackSteeringRateModel(const bark::commons::ParamsPtr& params)
+      : DynamicModel(params),
+        wheel_base_(params->GetReal("DynamicModel::WheelBase",
+                                    "Wheel base of vehicle [m]", 2.7)){
+  }
+
+  virtual ~SingleTrackSteeringRateModel() {}
+
+  State StateSpaceModel(const State& x, const Input& u) const {
+    State tmp(static_cast<int>(StateDefinition::MIN_STATE_SIZE ) + 1);
+    tmp << 1,
+        x(StateDefinition::VEL_POSITION) *
+            cos(x(StateDefinition::THETA_POSITION)),
+        x(StateDefinition::VEL_POSITION) *
+            sin(x(StateDefinition::THETA_POSITION)),
+        x(StateDefinition::VEL_POSITION) * tan(
+          x[5]) / wheel_base_,
```

Comment:
x[5] should also use StateDefinition

Reply:
The problem is that at the moment `StateDefinition::MIN_STATE_SIZE` is equal to `5`. So I don't think we can change that with major refactoring.

Reply:
Okay, although I don't like it :-)